packages/sakuli-legacy/src/context/legacy-lifecycle-hooks.class.ts

```diff
@@ -33,66 +33,50 @@ export class LegacyLifecycleHooks implements TestExecutionLifecycleHooks {
    */
   currentTest: Maybe<string> = null;
   uiOnly = false;
+  reuseBrowser = true;
 
   constructor(readonly builder: Builder) {}
 
   async onProject(project: Project) {
     const properties = project.objectFactory(LegacyProjectProperties);
     this.uiOnly = properties.isUiOnly();
-    if (!this.uiOnly) {
-      this.driver = createDriverFromProject(project, this.builder);
-      await this.driver.manage().window().maximize();
+    this.reuseBrowser = properties.isReuseBrowser();
+    if (!this.uiOnly && this.reuseBrowser) {
+      await this.createDriver(project);
     }
   }
 
-  async beforeExecution(
-    project: Project,
-    testExecutionContext: TestExecutionContext
-  ) {
+  async beforeExecution(project: Project, ctx: TestExecutionContext) {
     const properties = project.objectFactory(LegacyProjectProperties);
     const id = properties.testsuiteId
       ? properties.testsuiteId
       : project.rootDir.split(sep).pop();
     const warningTime = properties.testsuiteWarningTime || 0;
     const criticalTime = properties.testsuiteCriticalTime || 0;
-    testExecutionContext.startTestSuite({ id, warningTime, criticalTime });
+    ctx.startTestSuite({ id, warningTime, criticalTime });
```

Comment:
Why did you refactor a self explaining name like `testExecutionContext` to something cryptic like `ctx`? :thinking:

Reply:
We used `testExecutionContext` and `ctx` as a variable name for TestExecutionContext, so I refactored it all to `ctx` to stay consistent. In hindsight, it would have been better to use `testExecutionContext`